pkg/transform/kubernetes/populators_test.go

```diff
@@ -588,6 +588,53 @@ func TestPopulateEnvFrom(t *testing.T) {
 	}
 }
 
+func TestPopulateVolumes(t *testing.T) {
+	volumeClaims := []spec.VolumeClaim{{Name: "foo"}, {Name: "bar"}, {Name: "barfoo"}}
+	volumes := []api_v1.Volume{{Name: "foo"}}
+
+	// a volumeMount is defined but that is not there in volumeClaims
+	// neither it is in pod level volumes, so this should fail
+	failingContainers := []api_v1.Container{
+		{VolumeMounts: []api_v1.VolumeMount{{Name: "baz"}}},
+	}
+
+	if _, err := populateVolumes(failingContainers, volumeClaims, volumes); err == nil {
+		t.Errorf("should have failed but passed for volumeMount that" +
+			" does not exist.")
+	}
+
+	passingContainers := []api_v1.Container{
```

Comment:
I think it depends on e the situation and personal preference. What is more readable for one person, doesn't have to be always as readable to other.

Sometimes it makes a lot of sense to separate it, sometimes it doesn't.

I personally wouldn't probably do it like @surajssd  did it, but I don't see anything wrong with it either. Maybe just put `passingContainers` directly behind `failingContainers` so both definitions are together. But I'm fine with this.

I think there are more important things to discuss in this PR than this test ;-)